File: backend/app/routers/health.py
from datetime import UTC, datetime

from fastapi import APIRouter

from ..models.health import HealthResponse

router = APIRouter(tags=["health"])


@router.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    """Check the health status of the backend service.

    Returns the current status, version, and server timestamp.
    """
    return HealthResponse(
        status="healthy",
        version="0.1.0",
        timestamp=datetime.now(UTC),
    )

File: frontend/src/types/health.ts
/**
 * Health check response from the backend /api/health endpoint.
 *
 * This matches the backend's HealthResponse Pydantic model.
 */
export interface HealthStatus {
  /** Current health status (e.g., "healthy", "degraded", "unhealthy"). */
  status: string;

  /** Backend application version. */
  version: string;

  /** Number of seconds the backend has been running. */
  uptime_seconds?: number;

  /** ISO 8601 timestamp of when the health check was performed. */
  timestamp: string;
}
